ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/question_answering/QuestionAnsweringModel.java

```diff
@@ -47,50 +64,193 @@ public void warmUp(Predictor predictor, String modelId, MLModelConfig modelConfi
             throw new IllegalArgumentException("model id is null");
         }
 
+        // Initialize model type from config if not set
+        if (modelConfig != null) {
```

Comment:
`Initialize model type from config if not set`  this comment doesn't reflect what are we doing in `if (modelConfig != null) {`?

Reply:
what happens in `modelConfig` is null? also I wonder why unit tests are passing if modelConfig is null? Can we add a corresponding unit test?

Reply:
this seems more confusing..

Reply:
can we add comment on this?

Reply:
Sure I can add, I think this part code wasn't added by me. https://github.com/opensearch-project/ml-commons/commit/c560fcca6ee2e413e7ebc2503a2e64ad691e6e2b#diff-af4fa83b052ddffa01e7e652ec3dd13d13bbd4bce3a59143e405b9beb51af483R74-R82

Reply:
Updated comment here and also added comment in above test code.